src/webrtc.ts

```diff
@@ -526,10 +521,25 @@ export class WebrtcExtension extends EventEmitter {
       return ice;
     });
 
-    this.client._stanzaio.jingle.iceServers = iceServers;
+    this.setIceServers(iceServers);
+    if (!stunServers.services!.length) {
+      this.logger.info('No stun servers received, setting iceTransportPolicy to "relay"');
+      this.setIceTransportPolicy('relay');
+    } else {
+      this.setIceTransportPolicy('all');
+    }
+
     return iceServers;
   }
 
+  setIceServers (iceServers: any[]) {
+    this.client._stanzaio.jingle.iceServers = iceServers;
+  }
+
+  setIceTransportPolicy (policy: 'relay' | 'all') {
+    this.client._stanzaio.jingle.config.peerConnectionConfig!.iceTransportPolicy = policy;
```

Comment:
I don't like having to tunnel so deep into jingle, but there doesn't seem to be a better way to do this.